### ISSUE TITLE
ubuntu OS Error 

### ISSUE THREAD

**Issue by kylix3511** [galexrt/k8s-vagrant-multi-node#29]
Hi @galexrt  I got this error ...  have you seen this error before : 

``` 

kylix3511 >>>  BOX_OS=ubuntu KUBERNETES_VERSION=1.12.0 CLUSTER_NAME=k8s-local NO
DE_MEMORY_SIZE_GB=2 MASTER_MEMORY_SIZE_GB=3 NODE_CPUS=2 MASTER_CPUS=2 NODE_COUNT=2  make -j8 up 
if !(vagrant box list | grep -q generic/ubuntu1804); then \
		vagrant \
			box \
			add \
			--provider=virtualbox \
			generic/ubuntu1804; \
	else \
		vagrant box update --box=generic/ubuntu1804; \
	fi
Checking for updates to 'generic/ubuntu1804'
Latest installed version: 1.8.60
Version constraints: > 1.8.60
Provider: virtualbox
Box 'generic/ubuntu1804' (v1.8.60) is running the latest version.
vagrant up
NODE=1 vagrant up
NODE=2 vagrant up
Bringing machine 'node2' up with 'virtualbox' provider...
Bringing machine 'master' up with 'virtualbox' provider...
Bringing machine 'node1' up with 'virtualbox' provider...
==> node2: Importing base box 'generic/ubuntu1804'...
==> node1: Importing base box 'generic/ubuntu1804'...
==> master: Importing base box 'generic/ubuntu1804'...
==> master: Matching MAC address for NAT networking...
==> master: Checking if box 'generic/ubuntu1804' version '1.8.60' is up to date...
==> node1: Matching MAC address for NAT networking...
==> node2: Matching MAC address for NAT networking...
==> node1: Checking if box 'generic/ubuntu1804' version '1.8.60' is up to date...
==> node2: Checking if box 'generic/ubuntu1804' version '1.8.60' is up to date...
==> master: Setting the name of the VM: k8s-vagrant-multi-node_master_1549245554593_43065
==> node2: Setting the name of the VM: k8s-vagrant-multi-node_node2_1549245554970_91320
==> node1: Setting the name of the VM: k8s-vagrant-multi-node_node1_1549245554972_94381
==> master: Clearing any previously set network interfaces...
==> node1: Fixed port collision for 22 => 2222. Now on port 2200.
==> node1: Clearing any previously set network interfaces...
==> master: Preparing network interfaces based on configuration...
    master: Adapter 1: nat
    master: Adapter 2: hostonly
==> master: Forwarding ports...
    master: 8443 (guest) => 8443 (host) (adapter 1)
    master: 22 (guest) => 2222 (host) (adapter 1)
==> node1: Preparing network interfaces based on configuration...
    node1: Adapter 1: nat
    node1: Adapter 2: hostonly
==> node1: Forwarding ports...
==> master: Running 'pre-boot' VM customizations...
    node1: 22 (guest) => 2200 (host) (adapter 1)
==> node2: Fixed port collision for 22 => 2222. Now on port 2201.
==> node2: Clearing any previously set network interfaces...
==> master: Booting VM...
==> node2: Preparing network interfaces based on configuration...
    node2: Adapter 1: nat
    node2: Adapter 2: hostonly
==> node2: Forwarding ports...
==> node1: Running 'pre-boot' VM customizations...
    node2: 22 (guest) => 2201 (host) (adapter 1)
==> master: Waiting for machine to boot. This may take a few minutes...
==> node2: Running 'pre-boot' VM customizations...
==> node1: Booting VM...
    master: SSH address: 127.0.0.1:2222
    master: SSH username: vagrant
    master: SSH auth method: private key
==> node2: Booting VM...
==> node1: Waiting for machine to boot. This may take a few minutes...
==> node2: Waiting for machine to boot. This may take a few minutes...
    node1: SSH address: 127.0.0.1:2200
    node1: SSH username: vagrant
    node1: SSH auth method: private key
    node2: SSH address: 127.0.0.1:2201
    node2: SSH username: vagrant
    node2: SSH auth method: private key
    master: 
    master: Vagrant insecure key detected. Vagrant will automatically replace
    master: this with a newly generated keypair for better security.
    master: 
    master: Inserting generated public key within guest...
    master: Removing insecure key from the guest if it's present...
    master: Key inserted! Disconnecting and reconnecting using new SSH key...
    node1: 
    node1: Vagrant insecure key detected. Vagrant will automatically replace
    node1: this with a newly generated keypair for better security.
    node2: 
    node2: Vagrant insecure key detected. Vagrant will automatically replace
    node2: this with a newly generated keypair for better security.
    node1: 
    node1: Inserting generated public key within guest...
    node1: Removing insecure key from the guest if it's present...
    node1: Key inserted! Disconnecting and reconnecting using new SSH key...
    node2: 
    node2: Inserting generated public key within guest...
==> master: Machine booted and ready!
==> master: Checking for guest additions in VM...
    node2: Removing insecure key from the guest if it's present...
    master: The guest additions on this VM do not match the installed version of
    master: VirtualBox! In most cases this is fine, but in rare cases it can
    master: prevent things such as shared folders from working properly. If you see
    master: shared folder errors, please make sure the guest additions within the
    master: virtual machine match the version of VirtualBox you have installed on
    master: your host and reload your VM.
    master: 
    master: Guest Additions Version: 5.2.18
    master: VirtualBox Version: 6.0
==> master: Setting hostname...
    node2: Key inserted! Disconnecting and reconnecting using new SSH key...
==> master: Configuring and enabling network interfaces...
==> node1: Machine booted and ready!
==> node1: Checking for guest additions in VM...
    node1: The guest additions on this VM do not match the installed version of
    node1: VirtualBox! In most cases this is fine, but in rare cases it can
    node1: prevent things such as shared folders from working properly. If you see
    node1: shared folder errors, please make sure the guest additions within the
    node1: virtual machine match the version of VirtualBox you have installed on
    node1: your host and reload your VM.
    node1: 
    node1: Guest Additions Version: 5.2.18
    node1: VirtualBox Version: 6.0
==> node1: Setting hostname...
==> node2: Machine booted and ready!
==> node2: Checking for guest additions in VM...
    node2: The guest additions on this VM do not match the installed version of
    node2: VirtualBox! In most cases this is fine, but in rare cases it can
    node2: prevent things such as shared folders from working properly. If you see
    node2: shared folder errors, please make sure the guest additions within the
    node2: virtual machine match the version of VirtualBox you have installed on
    node2: your host and reload your VM.
    node2: 
    node2: Guest Additions Version: 5.2.18
    node2: VirtualBox Version: 6.0
==> node2: Setting hostname...
==> master: Rsyncing folder: /Users/rameshkumar/k8s/k8s-vagrant-multi-node/data/ubuntu-master/ => /data
==> node1: Configuring and enabling network interfaces...
==> node2: Configuring and enabling network interfaces...
==> master: Running provisioner: shell...
    master: Running: inline script
==> node1: Rsyncing folder: /Users/rameshkumar/k8s/k8s-vagrant-multi-node/data/ubuntu-node1/ => /data
==> node1: Running provisioner: shell...
    node1: Running: inline script
==> node2: Rsyncing folder: /Users/rameshkumar/k8s/k8s-vagrant-multi-node/data/ubuntu-node2/ => /data
==> master: Running provisioner: shell...
==> node2: Running provisioner: shell...
    master: Running: inline script
    master: ++ apt-get update
    node2: Running: inline script
==> node1: Running provisioner: shell...
    node1: Running: inline script
    node1: ++ apt-get update
==> node2: Running provisioner: shell...
    node2: Running: inline script
    node2: ++ apt-get update
    master: Hit:1 http://us.archive.ubuntu.com/ubuntu bionic InRelease
    master: Get:2 http://us.archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]
    master: Get:3 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]
    node2: Get:1 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]
    master: Get:4 http://us.archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]
    node2: Hit:2 http://us.archive.ubuntu.com/ubuntu bionic InRelease
    node1: Get:1 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]
    node2: Get:3 http://us.archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]
    node1: Hit:2 http://us.archive.ubuntu.com/ubuntu bionic InRelease
    master: Get:5 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [255 kB]
    node1: Get:3 http://us.archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]
    node1: Get:4 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [255 kB]
    node2: Get:4 http://us.archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]
    master: Get:6 http://us.archive.ubuntu.com/ubuntu bionic-updates/main i386 Packages [446 kB]
    node2: Get:5 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [255 kB]
    node2: Get:6 http://us.archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [512 kB]
    node1: Get:5 http://us.archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]
    node1: Get:6 http://security.ubuntu.com/ubuntu bionic-security/main i386 Packages [197 kB]
    master: Get:7 http://security.ubuntu.com/ubuntu bionic-security/main i386 Packages [197 kB]
    node1: Get:7 http://us.archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [512 kB]
    node1: Get:8 http://security.ubuntu.com/ubuntu bionic-security/main Translation-en [96.8 kB]
    master: Get:8 http://security.ubuntu.com/ubuntu bionic-security/main Translation-en [96.8 kB]
    master: Get:9 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [116 kB]
    node1: Get:9 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [116 kB]
    master: Get:10 http://security.ubuntu.com/ubuntu bionic-security/universe i386 Packages [114 kB]
    node1: Get:10 http://security.ubuntu.com/ubuntu bionic-security/universe i386 Packages [114 kB]
    master: Get:11 http://security.ubuntu.com/ubuntu bionic-security/universe Translation-en [66.2 kB]
    node1: Get:11 http://security.ubuntu.com/ubuntu bionic-security/universe Translation-en [66.2 kB]
    node2: Get:7 http://security.ubuntu.com/ubuntu bionic-security/main i386 Packages [197 kB]
    node2: Get:8 http://us.archive.ubuntu.com/ubuntu bionic-updates/main i386 Packages [446 kB]
    node2: Get:9 http://security.ubuntu.com/ubuntu bionic-security/main Translation-en [96.8 kB]
    node2: Get:10 http://security.ubuntu.com/ubuntu bionic-security/universe i386 Packages [114 kB]
    node1: Get:12 http://us.archive.ubuntu.com/ubuntu bionic-updates/main i386 Packages [446 kB]
    node2: Get:11 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [116 kB]
    master: Get:12 http://us.archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [512 kB]
    node2: Get:12 http://security.ubuntu.com/ubuntu bionic-security/universe Translation-en [66.2 kB]
    node2: Get:13 http://us.archive.ubuntu.com/ubuntu bionic-updates/main Translation-en [193 kB]
    node1: Get:13 http://us.archive.ubuntu.com/ubuntu bionic-updates/main Translation-en [193 kB]
    node1: Get:14 http://us.archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [724 kB]
    node2: Get:14 http://us.archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [724 kB]
    master: Get:13 http://us.archive.ubuntu.com/ubuntu bionic-updates/main Translation-en [193 kB]
    node1: Get:15 http://us.archive.ubuntu.com/ubuntu bionic-updates/universe i386 Packages [716 kB]
    master: Get:14 http://us.archive.ubuntu.com/ubuntu bionic-updates/universe i386 Packages [716 kB]
    node2: Get:15 http://us.archive.ubuntu.com/ubuntu bionic-updates/universe i386 Packages [716 kB]
    node1: Get:16 http://us.archive.ubuntu.com/ubuntu bionic-updates/universe Translation-en [183 kB]
    node1: Get:17 http://us.archive.ubuntu.com/ubuntu bionic-updates/multiverse Translation-en [3,452 B]
    master: Get:15 http://us.archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [724 kB]
    node1: Fetched 3,874 kB in 1min 20s (48.2 kB/s)
    node1: Reading package lists...
    node1: ++ apt-get install -y apt-transport-https curl software-properties-common ca-certificates
    node1: Reading package lists...
    node1: Building dependency tree...
    node1: 
    node1: Reading state information...
    node1: ca-certificates is already the newest version (20180409).
    node1: curl is already the newest version (7.58.0-2ubuntu3.5).
    node1: software-properties-common is already the newest version (0.96.24.32.7).
    node1: The following NEW packages will be installed:
    node1:   apt-transport-https
    node2: Get:16 http://us.archive.ubuntu.com/ubuntu bionic-updates/universe Translation-en [183 kB]
    node1: 0 upgraded, 1 newly installed, 0 to remove and 27 not upgraded.
    node1: Need to get 1,692 B of archives.
    node1: After this operation, 153 kB of additional disk space will be used.
    node1: Get:1 http://us.archive.ubuntu.com/ubuntu bionic-updates/universe amd64 apt-transport-https all 1.6.8 [1,692 B]
    node1: dpkg-preconfigure: unable to re-open stdin: No such file or directory
    node1: Fetched 1,692 B in 11s (157 B/s)
    node1: Selecting previously unselected package apt-transport-https.
    node1: (Reading database ... 
(Reading database ... 55%ase ... 5%
    node1: (Reading database ... 60%
    node1: (Reading database ... 65%
    node1: (Reading database ... 70%
    node1: (Reading database ... 75%
    node1: (Reading database ... 80%
    node1: (Reading database ... 85%
    node1: (Reading database ... 90%
    node1: (Reading database ... 95%
(Reading database ... 105289 files and directories currently installed.)
    node1: Preparing to unpack .../apt-transport-https_1.6.8_all.deb ...
    node1: Unpacking apt-transport-https (1.6.8) ...
    node1: Setting up apt-transport-https (1.6.8) ...
    node1: ++ sudo apt-key add -
    node1: ++ curl -fsSL https://download.docker.com/linux/ubuntu/gpg
    node1: Warning: apt-key output should not be parsed (stdout is not a terminal)
    node2: Get:17 http://us.archive.ubuntu.com/ubuntu bionic-updates/multiverse Translation-en [3,452 B]
    node1: OK
    node1: +++ lsb_release -cs
    node1: ++ add-apt-repository 'deb [arch=amd64] https://download.docker.com/linux/ubuntu    bionic    stable'
    node1: Hit:1 http://us.archive.ubuntu.com/ubuntu bionic InRelease
    node1: Hit:2 http://us.archive.ubuntu.com/ubuntu bionic-updates InRelease
    node1: Hit:3 http://security.ubuntu.com/ubuntu bionic-security InRelease
    node2: Fetched 3,874 kB in 1min 40s (38.6 kB/s)
    node2: Reading package lists...
    node1: Hit:4 http://us.archive.ubuntu.com/ubuntu bionic-backports InRelease
    node1: Get:5 https://download.docker.com/linux/ubuntu bionic InRelease [64.4 kB]
    node1: Get:6 https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages [3,695 B]
    node2: ++ apt-get install -y apt-transport-https curl software-properties-common ca-certificates
    node2: Reading package lists...
    node2: Building dependency tree...
    node2: 
    node2: Reading state information...
    node2: ca-certificates is already the newest version (20180409).
    node2: curl is already the newest version (7.58.0-2ubuntu3.5).
    node2: software-properties-common is already the newest version (0.96.24.32.7).
    node2: The following NEW packages will be installed:
    node2:   apt-transport-https
    node1: Fetched 68.1 kB in 2s (29.0 kB/s)
    node1: Reading package lists...
    node1: ++ curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg
    node1: ++ apt-key add -
    node1: Warning: apt-key output should not be parsed (stdout is not a terminal)
    node2: 0 upgraded, 1 newly installed, 0 to remove and 27 not upgraded.
    node2: Need to get 1,692 B of archives.
    node2: After this operation, 153 kB of additional disk space will be used.
    node2: Get:1 http://us.archive.ubuntu.com/ubuntu bionic-updates/universe amd64 apt-transport-https all 1.6.8 [1,692 B]
    node2: dpkg-preconfigure: unable to re-open stdin: No such file or directory
    node2: Fetched 1,692 B in 4s (408 B/s)
    node2: Selecting previously unselected package apt-transport-https.
    node2: (Reading database ... 
(Reading database ... 55%ase ... 5%
    node2: (Reading database ... 60%
    node2: (Reading database ... 65%
    node2: (Reading database ... 70%
    node2: (Reading database ... 75%
    node2: (Reading database ... 80%
    node2: (Reading database ... 85%
    node2: (Reading database ... 90%
    node2: (Reading database ... 95%
(Reading database ... 105289 files and directories currently installed.)
    node2: Preparing to unpack .../apt-transport-https_1.6.8_all.deb ...
    node2: Unpacking apt-transport-https (1.6.8) ...
    node2: Setting up apt-transport-https (1.6.8) ...
    node1: OK
    node1: ++ cat
    node1: ++ '[' -n 1.12.0 ']'
    node1: ++ KUBERNETES_PACKAGES='kubelet=1.12.0 kubeadm=1.12.0 kubectl=1.12.0'
    node1: ++ apt-get update
    node1: Hit:1 http://security.ubuntu.com/ubuntu bionic-security InRelease
    node1: Hit:2 http://us.archive.ubuntu.com/ubuntu bionic InRelease
    node1: Hit:3 http://us.archive.ubuntu.com/ubuntu bionic-updates InRelease
    node1: Hit:4 https://download.docker.com/linux/ubuntu bionic InRelease
    node1: Hit:5 http://us.archive.ubuntu.com/ubuntu bionic-backports InRelease
    master: Get:16 http://us.archive.ubuntu.com/ubuntu bionic-updates/universe Translation-en [183 kB]
    node2: ++ curl -fsSL https://download.docker.com/linux/ubuntu/gpg
    node2: ++ sudo apt-key add -
    node2: Warning: apt-key output should not be parsed (stdout is not a terminal)
    node2: OK
    node2: +++ lsb_release -cs
    node2: ++ add-apt-repository 'deb [arch=amd64] https://download.docker.com/linux/ubuntu    bionic    stable'
    master: Get:17 http://us.archive.ubuntu.com/ubuntu bionic-updates/multiverse Translation-en [3,452 B]
    node2: Hit:1 http://us.archive.ubuntu.com/ubuntu bionic InRelease
    node2: Hit:2 http://us.archive.ubuntu.com/ubuntu bionic-updates InRelease
    node2: Hit:3 http://us.archive.ubuntu.com/ubuntu bionic-backports InRelease
    master: Fetched 3,874 kB in 2min 5s (31.1 kB/s)
    master: Reading package lists...
    node2: Hit:4 http://security.ubuntu.com/ubuntu bionic-security InRelease
    node1: Get:6 https://packages.cloud.google.com/apt kubernetes-xenial InRelease [8,993 B]
    master: ++ apt-get install -y apt-transport-https curl software-properties-common ca-certificates
    master: Reading package lists...
    master: Building dependency tree...
    node2: Get:5 https://download.docker.com/linux/ubuntu bionic InRelease [64.4 kB]
    master: 
    master: Reading state information...
    master: ca-certificates is already the newest version (20180409).
    master: curl is already the newest version (7.58.0-2ubuntu3.5).
    master: software-properties-common is already the newest version (0.96.24.32.7).
    master: The following NEW packages will be installed:
    master:   apt-transport-https
    node1: Get:7 https://packages.cloud.google.com/apt kubernetes-xenial/main amd64 Packages [23.4 kB]
    node2: Get:6 https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages [3,695 B]
    node1: Fetched 32.4 kB in 15s (2,171 B/s)
    node1: Reading package lists...
    node2: Fetched 68.1 kB in 9s (7,180 B/s)
    node2: Reading package lists...
    node1: ++ apt-get install -y screen telnet docker-ce kubelet=1.12.0 kubeadm=1.12.0 kubectl=1.12.0
    node1: Reading package lists...
    node1: Building dependency tree...
    node1: Reading state information...
    node1: E
    node1: : 
    node1: Version '1.12.0' for 'kubelet' was not found
    node1: E
    node1: : 
    node1: Version '1.12.0' for 'kubeadm' was not found
    node1: E
    node1: : 
    node1: Version '1.12.0' for 'kubectl' was not found
    node1: ++ apt-mark hold kubelet kubeadm kubectl
    node1: kubelet set on hold.
    node1: kubeadm set on hold.
    node1: kubectl set on hold.
    node1: ++ swapoff -a
    node1: ++ sed -i '/swap/s/^/#/g' /etc/fstab
    node1: ++ echo 1
    node1: /tmp/vagrant-shell: line 23: /proc/sys/net/bridge/bridge-nf-call-iptables: No such file or directory
    node1: ++ '[' false '!=' false ']'
==> node1: Running provisioner: shell...
    node1: Running: inline script
    node1: ++ kubeadm reset -f
    node1: /tmp/vagrant-shell: line 2: kubeadm: command not found
    node1: ++ '[' -n '' ']'
    node1: ++ kubeadm join --discovery-token-unsafe-skip-ca-verification --token isif2q.k55krcvi1gu3sdhf 192.168.26.10:6443
    node1: /tmp/vagrant-shell: line 7: kubeadm: command not found
    node1: ++ echo 'Failed to kubeadm join'
    node1: Failed to kubeadm join
    node1: ++ exit 1
    node2: ++ curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg
    node2: ++ apt-key add -
    node2: Warning: apt-key output should not be parsed (stdout is not a terminal)
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
make[2]: *** [start-node-1] Error 1
make[2]: *** Waiting for unfinished jobs....
    master: 0 upgraded, 1 newly installed, 0 to remove and 27 not upgraded.
    master: Need to get 1,692 B of archives.
    master: After this operation, 153 kB of additional disk space will be used.
    master: Get:1 http://us.archive.ubuntu.com/ubuntu bionic-updates/universe amd64 apt-transport-https all 1.6.8 [1,692 B]
    master: dpkg-preconfigure: unable to re-open stdin: No such file or directory
    master: Fetched 1,692 B in 9s (189 B/s)
    master: Selecting previously unselected package apt-transport-https.
    master: (Reading database ... 
(Reading database ... 55%base ... 5%
    master: (Reading database ... 60%
    master: (Reading database ... 65%
    master: (Reading database ... 70%
    master: (Reading database ... 75%
    master: (Reading database ... 80%
    master: (Reading database ... 85%
    master: (Reading database ... 90%
    master: (Reading database ... 95%
(Reading database ... 105289 files and directories currently installed.)
    master: Preparing to unpack .../apt-transport-https_1.6.8_all.deb ...
    master: Unpacking apt-transport-https (1.6.8) ...
    master: Setting up apt-transport-https (1.6.8) ...
    master: ++ curl -fsSL https://download.docker.com/linux/ubuntu/gpg
    master: ++ sudo apt-key add -
    master: Warning: apt-key output should not be parsed (stdout is not a terminal)
    node2: OK
    node2: ++ cat
    node2: ++ '[' -n 1.12.0 ']'
    node2: ++ KUBERNETES_PACKAGES='kubelet=1.12.0 kubeadm=1.12.0 kubectl=1.12.0'
    node2: ++ apt-get update
    node2: Hit:1 http://security.ubuntu.com/ubuntu bionic-security InRelease
    node2: Hit:2 http://us.archive.ubuntu.com/ubuntu bionic InRelease
    node2: Hit:3 https://download.docker.com/linux/ubuntu bionic InRelease
    node2: Hit:4 http://us.archive.ubuntu.com/ubuntu bionic-updates InRelease
    node2: Hit:5 http://us.archive.ubuntu.com/ubuntu bionic-backports InRelease
    master: OK
    master: +++ lsb_release -cs
    master: ++ add-apt-repository 'deb [arch=amd64] https://download.docker.com/linux/ubuntu    bionic    stable'
    master: Hit:1 http://security.ubuntu.com/ubuntu bionic-security InRelease
    node2: Get:6 https://packages.cloud.google.com/apt kubernetes-xenial InRelease [8,993 B]
    node2: Get:7 https://packages.cloud.google.com/apt kubernetes-xenial/main amd64 Packages [23.4 kB]
    node2: Fetched 32.4 kB in 11s (2,954 B/s)
    node2: Reading package lists...
    master: Get:2 https://download.docker.com/linux/ubuntu bionic InRelease [64.4 kB]
    master: Hit:3 http://us.archive.ubuntu.com/ubuntu bionic InRelease
    node2: ++ apt-get install -y screen telnet docker-ce kubelet=1.12.0 kubeadm=1.12.0 kubectl=1.12.0
    node2: Reading package lists...
    node2: Building dependency tree...
    master: Get:4 https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages [3,695 B]
    node2: 
    node2: Reading state information...
    master: Hit:5 http://us.archive.ubuntu.com/ubuntu bionic-updates InRelease
    node2: E: Version '1.12.0' for 'kubelet' was not found
    node2: E: Version '1.12.0' for 'kubeadm' was not found
    node2: E: Version '1.12.0' for 'kubectl' was not found
    node2: ++ apt-mark hold kubelet kubeadm kubectl
    node2: kubelet set on hold.
    node2: kubeadm set on hold.
    node2: kubectl set on hold.
    node2: ++ swapoff -a
    node2: ++ sed -i '/swap/s/^/#/g' /etc/fstab
    node2: ++ echo 1
    node2: /tmp/vagrant-shell: line 23: /proc/sys/net/bridge/bridge-nf-call-iptables: No such file or directory
    node2: ++ '[' false '!=' false ']'
==> node2: Running provisioner: shell...
    master: Hit:6 http://us.archive.ubuntu.com/ubuntu bionic-backports InRelease
    node2: Running: inline script
    node2: ++ kubeadm reset -f
    node2: /tmp/vagrant-shell: line 2: kubeadm: command not found
    node2: ++ '[' -n '' ']'
    node2: ++ kubeadm join --discovery-token-unsafe-skip-ca-verification --token isif2q.k55krcvi1gu3sdhf 192.168.26.10:6443
    node2: /tmp/vagrant-shell: line 7: kubeadm: command not found
    node2: Failed to kubeadm join
    node2: ++ echo 'Failed to kubeadm join'
    node2: ++ exit 1
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
make[2]: *** [start-node-2] Error 1
    master: Fetched 68.1 kB in 14s (4,905 B/s)
    master: Reading package lists...
    master: ++ curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg
    master: ++ apt-key add -
    master: Warning: apt-key output should not be parsed (stdout is not a terminal)
    master: gpg: 
    master: no valid OpenPGP data found.
    master: ++ cat
    master: ++ '[' -n 1.12.0 ']'
    master: ++ KUBERNETES_PACKAGES='kubelet=1.12.0 kubeadm=1.12.0 kubectl=1.12.0'
    master: ++ apt-get update
    master: Hit:1 http://security.ubuntu.com/ubuntu bionic-security InRelease
    master: Hit:2 https://download.docker.com/linux/ubuntu bionic InRelease
    master: Hit:4 http://us.archive.ubuntu.com/ubuntu bionic InRelease
    master: Hit:5 http://us.archive.ubuntu.com/ubuntu bionic-updates InRelease
    master: Hit:6 http://us.archive.ubuntu.com/ubuntu bionic-backports InRelease
    master: Get:3 https://packages.cloud.google.com/apt kubernetes-xenial InRelease [8,993 B]
    master: Err:3 https://packages.cloud.google.com/apt kubernetes-xenial InRelease
    master:   The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 6A030B21BA07F4FB
    master: Reading package lists...
    master: W
    master: : 
    master: GPG error: https://packages.cloud.google.com/apt kubernetes-xenial InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 6A030B21BA07F4FB
    master: E
    master: : 
    master: The repository 'https://apt.kubernetes.io kubernetes-xenial InRelease' is not signed.
    master: ++ apt-get install -y screen telnet docker-ce kubelet=1.12.0 kubeadm=1.12.0 kubectl=1.12.0
    master: Reading package lists...
    master: Building dependency tree...
    master: Reading state information...
    master: E
    master: : 
    master: Unable to locate package kubelet
    master: E
    master: : 
    master: Unable to locate package kubeadm
    master: E
    master: : 
    master: Unable to locate package kubectl
    master: ++ apt-mark hold kubelet kubeadm kubectl
    master: E
    master: : 
    master: Unable to locate package kubelet
    master: E
    master: : 
    master: Unable to locate package kubeadm
    master: E
    master: : 
    master: Unable to locate package kubectl
    master: E
    master: : 
    master: No packages found
    master: ++ swapoff -a
    master: ++ sed -i '/swap/s/^/#/g' /etc/fstab
    master: ++ echo 1
    master: /tmp/vagrant-shell: line 23: /proc/sys/net/bridge/bridge-nf-call-iptables: No such file or directory
    master: ++ '[' false '!=' false ']'
==> master: Running provisioner: shell...
    master: Running: inline script
    master: ++ kubeadm reset -f
    master: /tmp/vagrant-shell: line 2: kubeadm: command not found
    master: ++ kubeadm init --kubernetes-version=1.12.0 --apiserver-advertise-address=192.168.26.10 --pod-network-cidr=10.244.0.0/16 --token isif2q.k55krcvi1gu3sdhf --token-ttl 0
    master: kubeadm init failed.
    master: /tmp/vagrant-shell: line 3: kubeadm: command not found
    master: ++ echo 'kubeadm init failed.'
    master: ++ exit 1
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
make[2]: *** [start-master] Error 1
make[1]: *** [start] Error 2
make: *** [up] Error 2
kylix3511 >>> 


```

**Comment by galexrt** [comment]
@kylix3511 I have merged a PR #30 which should fix those issues.
Please pull the latest `master` branch and check if the issue has been fixed for you now.

**Comment by galexrt** [comment]
Closing as no response for some time and during my further development and testing of the project I was not able to reproduce it.
